Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/GazeContingent/FixWindowBufferedSamples/EyeLink_FixWindowBufferedSamples.m

```diff
@@ -0,0 +1,538 @@
+    % If your text starts with "RECORDED BY " it will be available in DataViewer's Inspector window by clicking
+    % the EDF session node in the top panel and looking for the "Recorded By:" field in the bottom panel of the Inspector.
+    preambleText = sprintf('RECORDED BY Psychtoolbox demo %s session name: %s', mfilename, edfFile);
+    Eyelink('Command', 'add_file_preamble_text "%s"', preambleText);
+    
+    
+    %% STEP 2: SELECT AVAILABLE SAMPLE/EVENT DATA
+    % See EyeLinkProgrammers Guide manual > Useful EyeLink Commands > File Data Control & Link Data Control
+    
+    % Select which events are saved in the EDF file. Include everything just in case
+    Eyelink('Command', 'file_event_filter = LEFT,RIGHT,FIXATION,SACCADE,BLINK,MESSAGE,BUTTON,INPUT');
+    % Select which events are available online for gaze-contingent experiments. Include everything just in case
+    Eyelink('Command', 'link_event_filter = LEFT,RIGHT,FIXATION,SACCADE,BLINK,BUTTON,FIXUPDATE,INPUT');
+    % Select which sample data is saved in EDF file or available online. Include everything just in case
+    if ELsoftwareVersion > 3  % Check tracker version and include 'HTARGET' to save head target sticker data for supported eye trackers
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,HTARGET,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,HTARGET,STATUS,INPUT');
+    else
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,STATUS,INPUT');
+    end
+    
+    
+    %% STEP 3: OPEN GRAPHICS WINDOW
+    
+    % Open experiment graphics on the specified screen
+    if isempty(screenNumber)
+        screenNumber = max(Screen('Screens')); % Use default screen if none specified
+    end
+    [window, wRect] = Screen('OpenWindow', screenNumber, [128 128 128]); % Open graphics window
+    Screen('Flip', window);
+    % Return width and height of the graphics window/screen in pixels
+    [width, height] = Screen('WindowSize', window);
+    
+    
+    %% STEP 4: SET CALIBRATION SCREEN COLOURS; PROVIDE WINDOW SIZE TO EYELINK HOST & DATAVIEWER; SET CALIBRATION PARAMETERS; CALIBRATE
+    
+    % Provide EyeLink with some defaults, which are returned in the structure "el".
+    el = EyelinkInitDefaults(window);
+    % set calibration/validation/drift-check(or drift-correct) size as well as background and target colors. 
+    % It is important that this background colour is similar to that of the stimuli to prevent large luminance-based 
+    % pupil size changes (which can cause a drift in the eye movement data)
+    el.calibrationtargetsize = 3;% Outer target size as percentage of the screen
+    el.calibrationtargetwidth = 0.7;% Inner target size as percentage of the screen
+    el.backgroundcolour = [128 128 128];% RGB grey

```

Comment:
For normalized color range that would be [0.5 0.5 0.5] or for a simple gray just 0.5. This also allows to avoid the slight problem if a given graphics card + driver + operating system needs 128 or 127 for 50% gray.

Reply:
resolved in cf762efe7df1a9e7d29650dfcb83c66c0b571bd7